### ISSUE TITLE
Templating error in Create Custom Directory task

### ISSUE THREAD

**Issue by jedimt** [locp/ansible-role-cassandra#134]
**Describe the bug**
While running this playbook (which is most excellent btw) I'm getting the following templating error from Ansible referencing the locp.cassandra/tasks/configure.yml file. 

```
TASK [locp.cassandra : Create Custom Directory] **************************************************************************************************************************************************************
task path: /Users/aaron/.ansible/roles/locp.cassandra/tasks/configure.yml:3
Tuesday 15 June 2021  15:17:23 -0400 (0:00:00.062)       0:01:21.353 ********** 
fatal: [server-05.tme.nebulon.com]: FAILED! => {
    "msg": "template error while templating string: expected token ',', got 'integer'. String: {{ directory.0.spec.mode | default(0700) }}"
}
```

I used to have this set up on a Windows box, but moved to a Mac recently. While testing all my playbooks noticed this failure. 

**Expected behaviour**
The playbook should run to completion. 

**Screenshots**

**Versions (please complete the following information):**
- Ansible version: 2.11.0
- Role version: 1.6.8
- Python version: 3.8.2
ansible --version
ansible [core 2.11.0] 
  config file = /Users/aaron/Documents/github/demopod-ansible/ansible.cfg
  configured module search path = ['/Users/aaron/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/aaron/Library/Python/3.8/lib/python/site-packages/ansible
  ansible collection location = /Users/aaron/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/aaron/Library/Python/3.8/bin/ansible
  python version = 3.8.2 (default, Apr  8 2021, 23:19:18) [Clang 12.0.5 (clang-1205.0.22.9)]
  jinja version = 3.0.1
  libyaml = True

**Additional context**
Macbook running Big-Sur
Ansible installed via python `python -m pip install --user ansible`
No modifications made to the playbook.

group_vars file for Cassandra:
```
mount_dir: /data/db
cassandra_group: cassandra
cassandra_user: cassandra
java_packages: openjdk-8-jdk
cassandra_configuration:
  authenticator: PasswordAuthenticator
  cluster_name: MyCassandraCluster
  commitlog_directory: "{{mount_dir}}/cassandra/commitlog"
  commitlog_sync: periodic
  commitlog_sync_period_in_ms: 10000
  data_file_directories:
    - "{{ mount_dir }}/cassandra/data"
  endpoint_snitch: GossipingPropertyFileSnitch
  hints_directory: "{{mount_dir}}/cassandra/hints"
  listen_address: "{{ ansible_default_ipv4.address }}"
  # For a sensible value to set num_tokens to, please see
  # https://cassandra.apache.org/doc/latest/getting_started/production.html#tokens
  num_tokens: 4
  partitioner: org.apache.cassandra.dht.Murmur3Partitioner
  saved_caches_directory: "{{mount_dir}}//cassandra/saved_caches"
  seed_provider:
    - class_name: "org.apache.cassandra.locator.SimpleSeedProvider"
      parameters:
        # This sets the seed to the first node in the Ansible group called
        # cassandra.
        - seeds: "{{ hostvars[groups['cassandra'][0]]['ansible_default_ipv4'].address }}"
  start_native_transport: true
cassandra_configure_apache_repo: true
# Create an alternative directories structure for the Cassandra data.
# In this example, the will be a directory called /data owned by root
# with rwxr-xr-x permissions.  It will have a series of sub-directories
# all of which will be defaulted to being owned by the cassandra user
# with rwx------ permissions.
cassandra_directories:
  root:
    group: root
    mode: "0755"
    owner: root
    paths:
      - "{{mount_dir}}"
  data:
    paths:
      - "{{mount_dir}}/cassandra"
      - "{{mount_dir}}/cassandra/commitlog"
      - "{{mount_dir}}/cassandra/data"
      - "{{mount_dir}}/cassandra/hints"
      - "{{mount_dir}}/cassandra/saved_caches"
cassandra_regex_replacements:
  - path: cassandra-env.sh
    line: 'MAX_HEAP_SIZE="{{ cassandra_max_heapsize_mb }}M"'
    regexp: '^#MAX_HEAP_SIZE="4G"'
  - path: cassandra-env.sh
    line: 'HEAP_NEWSIZE="{{ cassandra_heap_new_size_mb }}M"'
    regexp: '^#HEAP_NEWSIZE="800M"'
  - path: cassandra-rackdc.properties
    line: 'dc=DC1'
    regexp: '^dc='
  - path: cassandra-rackdc.properties
    line: 'rack=RACK1'
    regexp: '^rack='
cassandra_repo_apache_release: 311x
```

**Comment by dallinb** [comment]
Hi @jedimt,

I noticed this as well just today from a failed job (see https://github.com/locp/ansible-role-cassandra/actions/runs/937633314 which has an identical error that you ).  This was working OK as recently as 1st of May.  Unfortunately we have a problem with our testing in that Ansible have not fixed https://github.com/ansible/galaxy/issues/2533 since October which is breaking our regression tests against the latest versions of Ansible.  Therefore this means we failed to pick this up as recently as we should have done.  However, we are still passing the tests (see https://github.com/locp/ansible-role-cassandra/actions/runs/940469682) which is being tested against Ansible 2.10.7.  Our last successful test was against Ansible 3.3.0.  It seems that our tests failed when 4.0.0 was released (https://pypi.org/project/ansible/#history).  I'll take a look and see if we can get this fixed, it may be that I need to have legacy support in version 1.X.Y of this role and move to a new 2.X.Y version for Ansible 4.0.0 and above.

**Comment by jedimt** [comment]
Thanks for the backstory! In the mean time I'm going to install a dedicated ansible box running 2.10 to get around this (hopefully).

**Comment by dallinb** [comment]
This may be relatively easy to fix (see #135).  Testing going OK so far so may try and get a new release out soon.  The fix seems compatible through 2.10.7 to 4.1.0.

**Comment by dallinb** [comment]
This issue will be closed when I merge #135 and shortly afterwards I will release version 1.6.9 of this role.  Please try that version of the role in your scenario.  If you still have problems, comment on this issue and I'll reopen it if necessary.

Please also consider completing the user survey at https://galaxy.ansible.com/locp/cassandra

**Comment by dallinb** [comment]
@jedimt Release [1.6.9](https://github.com/locp/ansible-role-cassandra/releases/tag/1.6.9) now available in https://galaxy.ansible.com/locp/cassandra
